src/transformers/trainer_tf.py

```diff
@@ -149,7 +152,14 @@ def _create_checkpoint_manager(self, max_to_keep: int = 5, load_model: bool = Tr
           load_model: if we want to start the training from the latest checkpoint.
         """
         ckpt = tf.train.Checkpoint(optimizer=self.optimizer, model=self.model)
-        self.model.ckpt_manager = tf.train.CheckpointManager(ckpt, PREFIX_CHECKPOINT_DIR, max_to_keep=max_to_keep)
+
+        logger.info("Saving model in {}".format(self.args.output_dir))
+
+        path = os.path.join(self.args.output_dir, PREFIX_CHECKPOINT_DIR)
+        os.makedirs(path, exist_ok=True)
+        logger.info("Saving model in {}".format(path))
+
+        self.model.ckpt_manager = tf.train.CheckpointManager(ckpt, path, max_to_keep=max_to_keep)
```

Comment:
This behavior should be in sync with the Pytorch version, so would be better to keep it as it is. The main reason is because when someone wants to upload his model in the hub, the checkpoints should not be uploaded in same time :)

Reply:
Oh, did'nt think about that. But isnt that the behavior of the pytorch trainer? The only reason I started to touch that was because it would fail with a missing directory.

`
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.FailedPreconditionError: checkpoint/ckpt-1_temp_3bb3a094412148b598753eb71c5b94e0; Not a directory [Op:SaveV2]
`

This is from the PT version:
`
output_dir = os.path.join(self.args.output_dir, f"{PREFIX_CHECKPOINT_DIR}-{global_step}")
`
Which gives a directory that looks like this
```
models_syn/swag_base$ ls

checkpoint-1000   checkpoint-2000  checkpoint-4500  checkpoint-6500  checkpoint-9000    special_tokens_map.json
checkpoint-10000  checkpoint-2500  checkpoint-500   checkpoint-7000  checkpoint-9500    tokenizer_config.json
checkpoint-10500  checkpoint-3000  checkpoint-5000  checkpoint-7500  config.json        training_args.bin
checkpoint-11000  checkpoint-3500  checkpoint-5500  checkpoint-8000  eval_results.txt   vocab.txt
checkpoint-1500   checkpoint-4000  checkpoint-6000  checkpoint-8500  pytorch_model.bin
```

So maybe switch to that?

Reply:
The error you get is surprising... Can you give me a tiny piece of code to reproduce it? I tried with several trainers and I cannot make it fails.

Reply:
Working on it. I only did test it with my MPCtrainer.

Reply:
Example with the glue trainer:
```
python run_tf_glue.py --task_name mrpc --model_name_or_path bert-base-cased --output_dir model --max-seq-length 128 --num_train_epochs 1 --per_gpu_train_batch_size 16 --per_gpu_eval_batch_size 32 --do_train --logging_dir logs --mode sequence-classification --logging_steps 10 --evaluate_during_training --save_steps 10
```
```
05/11/2020 18:22:18 - INFO - transformers.trainer_tf -   Epoch 1 Step 10 Train Loss 0.5562
05/11/2020 18:22:21 - INFO - transformers.trainer_tf -   Saving checkpoint for step 10 at checkpoint/ckpt-1
05/11/2020 18:22:29 - INFO - transformers.trainer_tf -   Epoch 1 Step 20 Train Loss 0.6230
05/11/2020 18:22:32 - INFO - transformers.trainer_tf -   Saving checkpoint for step 20 at checkpoint/ckpt-2
05/11/2020 18:22:41 - INFO - transformers.trainer_tf -   Epoch 1 Step 30 Train Loss 0.4781
05/11/2020 18:22:43 - INFO - transformers.trainer_tf -   Saving checkpoint for step 30 at checkpoint/ckpt-3
05/11/2020 18:22:52 - INFO - transformers.trainer_tf -   Epoch 1 Step 40 Train Loss 0.5987
05/11/2020 18:22:55 - INFO - transformers.trainer_tf -   Saving checkpoint for step 40 at checkpoint/ckpt-4
05/11/2020 18:23:04 - INFO - transformers.trainer_tf -   Epoch 1 Step 50 Train Loss 0.6039
05/11/2020 18:23:07 - INFO - transformers.trainer_tf -   Saving checkpoint for step 50 at checkpoint/ckpt-5
05/11/2020 18:23:15 - INFO - transformers.trainer_tf -   Epoch 1 Step 60 Train Loss 0.5790
05/11/2020 18:23:18 - INFO - transformers.trainer_tf -   Saving checkpoint for step 60 at checkpoint/ckpt-6
05/11/2020 18:23:27 - INFO - transformers.trainer_tf -   Epoch 1 Step 70 Train Loss 0.5019
05/11/2020 18:23:30 - INFO - transformers.trainer_tf -   Saving checkpoint for step 70 at checkpoint/ckpt-7
05/11/2020 18:23:39 - INFO - transformers.trainer_tf -   Epoch 1 Step 80 Train Loss 0.5242
05/11/2020 18:23:42 - INFO - transformers.trainer_tf -   Saving checkpoint for step 80 at checkpoint/ckpt-8
05/11/2020 18:23:51 - INFO - transformers.trainer_tf -   Epoch 1 Step 90 Train Loss 0.3365
05/11/2020 18:23:54 - INFO - transformers.trainer_tf -   Saving checkpoint for step 90 at checkpoint/ckpt-9
05/11/2020 18:24:03 - INFO - transformers.trainer_tf -   Epoch 1 Step 100 Train Loss 0.4605
05/11/2020 18:24:05 - INFO - transformers.trainer_tf -   Saving checkpoint for step 100 at checkpoint/ckpt-10
05/11/2020 18:24:14 - INFO - transformers.trainer_tf -   Epoch 1 Step 110 Train Loss 0.5719
05/11/2020 18:24:17 - INFO - transformers.trainer_tf -   Saving checkpoint for step 110 at checkpoint/ckpt-11
05/11/2020 18:24:22 - INFO - transformers.trainer_tf -   Saving model in model
05/11/2020 18:24:22 - INFO - transformers.configuration_utils -   Configuration saved in model/config.json
05/11/2020 18:24:23 - INFO - transformers.modeling_tf_utils -   Model weights saved in model/tf_model.h5
```

Reply:
Works on glue. Weird. Guess it something on my end.

Reply:
I think it might be also related to colab, I don't know, I don't use that much colab. For TPUs, I just tried one GLUE dataset.